app/views/results/show.html.slim

```diff
@@ -5,7 +5,10 @@
     = t(@check.controlled? ? ".controlled_qualifies" : ".likely_to_qualify")
 
 - content_for :back do
-  = link_to t("generic.back"), check_answers_path(assessment_code: params[:assessment_code]), class: "govuk-back-link"
+  - if FeatureFlags.enabled?(:ee_banner, @check.session_data) && @early_result_type == "gross_income"
```

Comment:
looks good - might not be necessary at the min but to future proof this for other early result type checks, you could just check for the presence of `early_result_type` (rather than tying it to gross_income specifically), as the behaviour of going back to the previous step should be the same for all early results

Reply:
and early_result_type should only exist if they never saw 'Check Answers' (the place where we clear it out from the session)

Reply:
ooooh actually it's wouldn't exist for short checks e.g. passported! So maybe this is the best way for now, or maybe the condition is check for early result and check it isn't a 'short check'...will leave to you!

Reply:
I have changed it to:
`if FeatureFlags.enabled?(:ee_banner, @check.session_data) && @early_result_type`